Bottom: 8706747d8841cecefce6c394ef08bec016dd0348
Top:    50835e142dc27f9d7b745b34a0d7c1416c75a0b7
Author: Sean Hefty <sean.hefty@intel.com>
Date:   2012-08-07 11:08:30 -0700

support for debug prints


---

diff --git a/src/preload.c b/src/preload.c
index 53f8543..1264594 100644
--- a/src/preload.c
+++ b/src/preload.c
@@ -48,6 +48,7 @@
 #include <netinet/tcp.h>
 #include <unistd.h>
 #include <semaphore.h>
+#include <stdio.h>
 
 #include <rdma/rdma_cma.h>
 #include <rdma/rdma_verbs.h>
@@ -98,6 +99,8 @@ static int rq_size;
 static int sq_inline;
 static int fork_support;
 
+static FILE *fout;
+
 enum fd_type {
 	fd_normal,
 	fd_rsocket,
@@ -286,6 +289,7 @@ static void init_preload(void)
 	rs.getsockopt = dlsym(RTLD_DEFAULT, "rgetsockopt");
 	rs.fcntl = dlsym(RTLD_DEFAULT, "rfcntl");
 
+	fout = fopen("rs-out.txt", "w+");
 	getenv_options();
 	init = 1;
 out:
@@ -968,12 +972,18 @@ int __fxstat64(int ver, int socket, struct stat64 *buf)
 {
 	int fd, ret;
 
+	init_preload();
+	fprintf(fout, "fxstat64 socket %d - fd %d type %d\n",
+		socket, fd_getd(socket), fd_gett(socket));
 	if (fd_get(socket, &fd) == fd_rsocket) {
 		ret = real.fxstat64(ver, socket, buf);
+		fprintf(fout, "fxstat64 - rsocket %d\n", ret);
 		if (!ret)
 			buf->st_mode = (buf->st_mode & ~S_IFMT) | __S_IFSOCK;
 	} else {
 		ret = real.fxstat64(ver, fd, buf);
+		fprintf(fout, "fxstat64 - normal %d\n", ret);
 	}
+	fflush(fout);
 	return ret;
 }
